sklearn/base.py

```diff
@@ -1087,3 +1088,33 @@ def is_outlier_detector(estimator):
         True if estimator is an outlier detector and False otherwise.
     """
     return getattr(estimator, "_estimator_type", None) == "outlier_detector"
+
+
+def _fit_context(*, skip_nested_validation=True):
+    """Decorator to run the fit methods of estimators within context managers.
+
+    Parameters
+    ----------
+    skip_nested_validation : bool, default=True
+        Whether to skip the validation of parameters of inner estimators or functions
+        called during fit.
+
+    Returns
+    -------
+    decorated_fit : method
+        The decorated fit method.
+    """
+
+    def decorator(fit_method):
+        @functools.wraps(fit_method)
+        def wrapper(estimator, *args, **kwargs):
+
+            if not get_config()["skip_parameter_validation"]:
+                estimator._validate_params()
+
+            with config_context(skip_parameter_validation=skip_nested_validation):
```

Comment:
Assuming that the meta-estimator will set `skip_nested_validation=False`, the following will always force the `base_estimator` to validate.

```python
with config_context(skip_parameter_validation=True):
    MetaEstimator(base_estimator).fit(X, y)
```

I expect that the above would turn off all validation, even in the `base_estimator`. With that in mind, I think we still need to consider the global config:

```python
def _fit_context(*, prefer_skip_nested_validation=True):

    ...

    def wrapper(estimator, *args, **kwargs):
        global_skip_parameter_validation = get_config()["skip_parameter_validation"]
        if not global_skip_parameter_validation:
            estimator._validate_params()

	skip_nested_validation = (
            global_skip_parameter_validation or prefer_skip_nested_validation
        )
        with config_context(skip_parameter_validation=skip_nested_validation):
            return fit_method(estimator, *args, **kwargs)
```

Here the decorator will give a preference, but still consider the global configuration.

Reply:
Good point, we want to skip if either validation is skipped by the config_context or if nested validation is skipped.
I added a test for that.